pybleau/app/ui/dataframe_plot_manager_view.py

```diff
@@ -214,27 +213,38 @@ def _new_plot_button_fired(self):
         selector = PlotTypeSelector(
             view_klass=self.view_klass,
             num_containers=self.model.canvas_manager.num_container_managers,
-            container_idx=AUTO_TARGET_CONTAINER
+            container_idx=AUTO_TARGET_CONTAINER,
+            _plot_types=self.model.plot_types
         )
         ui = selector.edit_traits(kind="modal")
         if not ui.result:
             return
 
         plot_type = selector.plot_type
 
-        next_plot_num = len(self.model.contained_plots) + 1
-        if plot_type.startswith("Multi"):
-            new_plot_default_title = "Plot {i}"
+        if plot_type in list(self.model.custom_configs.keys()) and \
+                self.model.template_interactor is not None:
```

Comment:
I think the second `if` should be tested inside the first `if` and log the issue and raise and exception if the interactor is `None` since it's an inconsistent state. Otherwise, the code will go to the `else` clause and break there in a non-meaningful way.

Reply:
Great find. Thanks.

> I think the second `if` should be tested inside the first `if` and log the issue and raise and exception if the interactor is `None` since it's an inconsistent state. Otherwise, the code will go to the `else` clause and break there in a non-meaningful way.